### PR TITLE
Providing common state for extra formatter

### PR DESCRIPTION
**Pull request by fvgh**
Refactored EclipseFormatterStep. Extracted inner class to provide reusability in future features.

### CODE REVIEW

**Comment by fvgh** [comment]
@nedtwigg
This version of the ExtFormatterState is basically the complete opposite form the version I started off with. Based on [your comments](https://github.com/diffplug/spotless/compare/master...fvgh:feature/groovy#commitcomment-20969829)
I tried to reduced it to the very minimum currently needed.

File: lib-extra/src/main/java/com/diffplug/spotless/extra/ExtFormatterState.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	private String formatterMethod;
+
+	private ExtFormatterState(
+			final JarState jarState,
+			final String className,
+			final FileSignature configFileSignature) {
+		this.jarState = jarState;
+		this.configFileSignature = configFileSignature;
+		this.className = className;
+		this.formatterMethod = DEFAULT_FORMATTER_METHOD;
```
Comment (nedtwigg) [inline lib-extra/src/main/java/com/diffplug/spotless/extra/ExtFormatterState.java]: It's not possible to change the value of the formatterMethod field.  Why is it an instance method?

File: lib-extra/src/main/java/com/diffplug/spotless/extra/java/EclipseFormatterStep.java [hunk 2]
```diff
@@ -38,64 +31,23 @@ private EclipseFormatterStep() {}
...
 	private static final String FORMATTER_CLASS = "com.diffplug.gradle.spotless.java.eclipse.EclipseFormatterStepImpl";
-	private static final String FORMATTER_METHOD = "format";
-
-	/** Creates a formatter step for the given version and settings file.
-	 * Formatter steps based on property configuration should support zero (default configuration)
-	 * to many files. Use {@link #create(Iterable, Provisioner)} instead.*/
-	@Deprecated
-	public static FormatterStep create(File settingsFile, Provisioner provisioner) {
-		return create(Arrays.asList(settingsFile), provisioner);
-	}
```
Comment (nedtwigg) [inline lib-extra/src/main/java/com/diffplug/spotless/extra/java/EclipseFormatterStep.java]: We want to keep the @Deprecated methods.

File: lib-extra/src/main/java/com/diffplug/spotless/extra/java/EclipseFormatterStep.java [hunk 3]
```diff
@@ -38,64 +31,23 @@ private EclipseFormatterStep() {}
...
 	}
 
-	/** Creates a formatter step for the given version and settings file.
-	 * Formatter steps based on property configuration should support zero (default configuration)
-	 * to many files. Use {@link #create(String, Iterable, Provisioner)} instead.*/
-	@Deprecated
-	public static FormatterStep create(String version, File settingsFile, Provisioner provisioner) {
-		return create(version, Arrays.asList(settingsFile), provisioner);
-	}
-
```
Comment (nedtwigg) [inline lib-extra/src/main/java/com/diffplug/spotless/extra/java/EclipseFormatterStep.java]: Why removing the @Deprecated method?

**Comment by fvgh** [comment]
Sorry about the deprecates. They should remain, I had not final look at the diff. My mistake.
About your initial comment:
Just thought that the code may otherwise be copied multiple times.
But comparing both versions again, I agree. I withdraw the proposal and stay with the original code.

**Pull request closed without merging**